src/api/referral.ts

```diff
@@ -0,0 +1,323 @@
+import express, { Request, Response } from 'express';
+import { referralService } from '../services/referral/index.ts';
+import { errorHandlerAsync } from '../middleware/errorHandler.ts';
+import { validateBody } from '../middleware/validator.ts';
+import { ApiError, successResponse } from '../middleware/types/errors.ts';
+
+const router = express.Router();
+
+// Generate referral code for a wallet
+router.post(
+  '/generate-code',
+  errorHandlerAsync(async (req: Request, res: Response) => {
+    const { walletAddress } = req.body;
+
+    if (!walletAddress) {
+      throw ApiError.badRequest('Wallet address is required');
+    }
+
+    const referralCode = await referralService.generateReferralCode(walletAddress);
+    const referralLink = `${process.env.FRONTEND_URL || 'https://clones.sol'}/ref/${referralCode}`;
```

Comment:
It is preferable to set the default to https://clones-ai.com/ because clones.sol is not accessible by default.